cfme/tests/intelligence/reports/test_crud.py

```diff
@@ -240,3 +241,21 @@ def test_import_report(appliance):
     response, = appliance.rest_api.collections.reports.action.execute_action("import", data)
     assert_response(appliance)
     assert response['message'] == 'Skipping Report (already in DB): [{}]'.format(menu_name)
+
+
+@pytest.mark.sauce
+@pytest.mark.tier(3)
+@test_requirements.report
+def test_reports_delete_saved_report(custom_report_values, appliance):
+    """This test case deletes reports from the Saved Reports list.
+    """
+    report = appliance.collections.reports.create(**custom_report_values)
+    report.queue(wait_for_finish=True)
+    view = navigate_to(appliance.collections.saved_reports, 'All')
+    # iterates through every row and checks if the 'Name' column matches the given value
+    for row in view.table.rows():
+        if row.name.read() == report.title:
+            row[0].check()
+    view.configuration.item_select(
+        item='Delete selected Saved Reports', handle_alert=True)
+    report.delete()
```

Comment:
Required: Add a finalizer for this deletion
request.addfinalizer(report.delete)
and move it right aftre report create